libs/injection/bbhnet/injection/priors.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+from bilby.core.prior import (
+    ConditionalPowerLaw,
+    ConditionalPriorDict,
+    Constraint,
+    Cosine,
+    PowerLaw,
+    PriorDict,
+    Sine,
+    Uniform,
+)
+from bilby.gw.prior import UniformComovingVolume, UniformSourceFrame
+
+
+def prior_selector(prior_name):
+
+    if prior_name == "nonspin_bbh":
+        return nonspin_bbh()
+    elif prior_name == "end_o3_ratesandpops":
+        return end_o3_ratesandpops()
+    else:
+        raise ValueError(f"Prior name {prior_name} is not an option")
```

Comment:
I don't think you need this prior_selector. When we need to specify a Prior in a config, we can annotate that in the function args as a `Callable`, and in our `typeo` config pass the python path, e.g `"bbhnet.injection.priors.nonspin_bbh"`

Reply:
I think this is exactly right. I think there's maybe even a case for having this hardcoded into a project, e.g. `from bbhnet.injection.priors import nonspin_bbh`, but if we think this is something we'll want to have some flexibility with at the project level this works.

Reply:
The other alternative is to just construct these dicts at the outermost level of `priors.py` and make them available via `priors` dict of dicts, i.e.:
```python
nonspin_bbh = ...
end_o3_ratesandprops = ...
priors = {
    "nonspin_bbh": nonspin_bbh,
    "end_o3_ratesandprops": end_o3_ratesandprops
}
```
But if these are expensive to construct and will add non-trivial import time this probably isn't the right way to go.

Reply:
Good stuff, I was hoping there was a better solution than this function. I'll try out both methods and see if there's a difference.